lib/puppet/type/rds_instance.rb

```diff
@@ -0,0 +1,181 @@
+
+The name of the database to create when the DB instance is created. If this parameter is not specified, no database is created in the DB instance.
+
+Constraints:
+
+Must contain 1 to 63 alphanumeric characters
+Must begin with a letter or an underscore. Subsequent characters can be letters, underscores, or digits (0-9).
+Cannot be a word reserved by the specified database engine
+Oracle
+
+The Oracle System ID (SID) of the created DB instance.
+
+Default: ORCL
+
+Constraints:
+
+Cannot be longer than 8 characters
+SQL Server
+
+Not applicable. Must be null.'
+  end
+
+  newparam(:security_groups, :array_matching => :all) do
+    desc 'the security groups to associate the instance'
+  end
+
+  newproperty(:tags) do
+    desc 'the tags for the instance'
+  end
+
+  newproperty(:region) do
+    desc 'the region in which to launch the instance'
+    validate do |value|
+      fail 'region should not contain spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:db_instance_class) do
+    desc 'the type to use for the instance (mysql, | postgres)'
+    validate do |value|
+      fail 'db_instance_class should not contain spaces' if value =~ /\s/
+      fail 'db_instance_class should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:availability_zone) do
+    desc 'the availability zone in which to place the instance'
+    validate do |value|
+      fail 'availability_zone should not contain spaces' if value =~ /\s/
+      fail 'availability_zone should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:engine) do
+    desc 'the type of Database for the instance( mysql, postgres, etc...)'
+    validate do |value|
+      fail 'engine type should not contains spaces' if value =~ /\s/
+      fail 'engine should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:engine_version) do
+    desc 'the version of Database for the instance'
+    validate do |value|
+      fail 'engine_version type should not contains spaces' if value =~ /\s/
+      fail 'engine_version should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:allocated_storage) do
+    desc 'The size of the DB'
+    validate do |value|
+      fail 'allocated_storage type should not contains spaces' if value =~ /\s/
+      fail 'allocated_storage should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:license_model) do
+    desc 'the license for the instance (iValid values: license-included | bring-your-own-license | general-public-license)'
+    validate do |value|
+      fail 'license_model type should not contains spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:storage_type) do
+    desc 'The storage stype for the DB (Valid values: gp | io1  *Note: If you specify io1, you must also include a value for the Iops parameter)'
+    validate do |value|
+      fail 'storage_type type should not contains spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:iops) do
+    desc 'The IOPS stype for the DB (minimum 1000)'
```

Comment:
From reading the docs, by default this is 0, and that will set your RDS as standard storage rather than the Provisioned IOPS. So I'll probably remove the comment about minimum